std-bits/table/src/main/java/org/enso/table/excel/ExcelHeaders.java

```diff
@@ -62,7 +61,27 @@ private static String[] readRowAsHeaders(
     for (int col = startCol; col <= currentEndCol; col++) {
       Cell cell = row.get(col);
 
-      String cellText = cell == null ? "" : formatter.formatCellValue(cell);
+      var resultType =
+          cell.getCellType() == CellType.FORMULA
+              ? cell.getCachedFormulaResultType()
+              : cell.getCellType();
+      String cellText =
+          switch (resultType) {
+            case STRING -> cell.getRichStringCellValue().getString();
+            case BOOLEAN -> cell.getBooleanCellValue() ? "TRUE" : "FALSE";
+            case BLANK -> "";
+            case ERROR -> FormulaError.forInt(cell.getErrorCellValue()).getString();
+            case NUMERIC -> {
+              var format = ExcelNumberFormat.from(cell, null);
+              var value = cell.getNumericCellValue();
+              yield format == null
+                  ? Double.toString(value)
+                  : formatter.formatRawCellContents(value, format.getIdx(), format.getFormat());
+            }
+            default ->
+            // Fallback to the formatted value
+            formatter.formatCellValue(cell);
```

Comment:
This logic is a bit similar to `getCellValue`. Could we merge those?

IMHO these two flows should yield the same values:
```
t1 = Data.read (... headers=False) . use_first_row_as_headers
t2 = Data.read (... headers=True)
```

While currently `t1` will go through `getCellValue` and `t2` will go through this. If both methods yield the same result - we probably can delegate and not repeat the implementation. If they yield different results (which is what I suspect), should such inconsistency be allowed?

Reply:
Not quite the same result as preserves formatting when read directly to headers. Will see if can use `getCellValue` more.

Reply:
Ok, I see.

In that case I guess different logic for the headers may make sense. But then I guess I'd at least extract this small piece of logic into a separate method, instead of keeping it inside of this, now relatively large, loop.